ads/inabox/position-observer.js

```diff
@@ -14,9 +14,82 @@
  * limitations under the License.
  */
 
-import {layoutRectLtwh, LayoutRectDef} from '../../src/layout-rect';
+import {
+  layoutRectLtwh,
+  LayoutRectDef,
+  layoutRectFromDomRect,
+} from '../../src/layout-rect';
 import {Observable} from '../../src/observable';
 import {throttle} from '../../src/utils/rate-limit';
+import {
+  PosObViewportInfoDef,
+} from '../../src/service/position-observer/position-observer-viewport-info';
+
+
+/**
+ * @implements {PosObViewportInfoDef}
```

Comment:
Haha I am really bad at naming. 
changed to `PosObHostInterfaceDef`, then `PosObAmpdocHostInterface` `PosObInaboxHostInterface` `PosObNonAmpHostInterface`